### PR TITLE
net: sockets: getaddrinfo: Minor refactoring

### PR DESCRIPTION
**Pull request by stig-bjorlykke**
Minor refactoring in getaddrinfo() to make the code easier to read and to make handling IPv4 and IPv6 support more equal.

- Move common wait and error handling code to exec_query()
- Use the same check for CONFIG_NET_IPV4 and CONFIG_NET_IPV6
- Add extra sanity check for family before exec_query()

Fix issue with setting port number for all DNS servers.

### CODE REVIEW

File: subsys/net/lib/sockets/getaddrinfo.c [hunk 1]
```diff
@@ -107,14 +107,42 @@ static int exec_query(const char *host, int family,
...
+			st = DNS_EAI_AGAIN;
+		} else {
+			st = ai_state->status;
+		}
+	} else if (ret == -EPFNOSUPPORT) {
+		/* If we are returned -EPFNOSUPPORT then that will indicate
+		 * wrong address family type queried. Check that and return
+		 * DNS_EAI_ADDRFAMILY and set errno to EINVAL.
+		 */
+		errno = EINVAL;
```
Comment (rlubos) [inline subsys/net/lib/sockets/getaddrinfo.c:138]: I know it's a copy-paste, but `errno` only needs to be fixed for `DNS_EAI_SYSTEM`, perhaps a good occasion to fix that.

**Review by rlubos — COMMENTED** [review]
Generally looks reasonable. I don't follow though, which part of the PR `Fix issue with setting port number for all DNS servers.` refers to?

File: subsys/net/lib/sockets/getaddrinfo.c (hunk 1 above)
Comment (stig-bjorlykke) [reply subsys/net/lib/sockets/getaddrinfo.c:138]: Right, I'll remove this.

**Comment by stig-bjorlykke** [comment]
> Generally looks reasonable. I don't follow though, which part of the PR `Fix issue with setting port number for all DNS servers.` refers to?

It's the last change from updating port number for the last entry to updating port number of all entries. This was introduced in the combination of fix for #15197 and #16453.

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**